File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/converter/AndroidAddressToWalletAddressConverter.java
package com.worldventures.dreamtrips.wallet.domain.converter;

import android.location.Address;

import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.ImmutableWalletAddress;
import com.worldventures.dreamtrips.wallet.domain.entity.lostcard.WalletAddress;

import io.techery.mappery.MapperyContext;


public class AndroidAddressToWalletAddressConverter implements Converter<Address, WalletAddress> {
   @Override
   public Class<Address> sourceClass() {
      return Address.class;
   }

   @Override
   public Class<WalletAddress> targetClass() {
      return WalletAddress.class;
   }

   @Override
   public WalletAddress convert(MapperyContext mapperyContext, Address address) {
      return ImmutableWalletAddress.builder()
            .addressLine(address.getAddressLine(0))
            .countryName(address.getCountryName())
            .adminArea(address.getAdminArea())
            .postalCode(address.getPostalCode())
            .build();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/AndroidAddressToWalletAddressConverter.kt
package com.worldventures.wallet.domain.converter

import android.location.Address

import com.worldventures.core.converter.Converter
import com.worldventures.wallet.domain.entity.lostcard.WalletAddress

import io.techery.mappery.MapperyContext

class AndroidAddressToWalletAddressConverter : Converter<Address, WalletAddress> {

   override fun sourceClass(): Class<Address> = Address::class.java

   override fun targetClass(): Class<WalletAddress> = WalletAddress::class.java

   override fun convert(context: MapperyContext, source: Address): WalletAddress
         = WalletAddress(source.getAddressLine(0), source.countryName, source.adminArea, source.postalCode)
}
